contracts/mainnet/connectors/morpho-blue/helpers.sol

```diff
@@ -0,0 +1,193 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "./interface.sol";
+import "../../common/stores.sol";
+import "../../common/basic.sol";
+import "../../common/interfaces.sol";
+
+abstract contract Helpers is Stores, Basic {
+	IMorpho public constant MORPHO_BLUE =
+		IMorpho(0x777777c9898D384F785Ee44Acfe945efDFf5f3E0); // TODO: Update
+
+	uint256 internal constant MARKET_PARAMS_BYTES_LENGTH = 5 * 32;
+
+	/// @dev The number of virtual assets of 1 enforces a conversion rate between shares and assets when a market is
+	/// empty.
+	uint256 internal constant VIRTUAL_ASSETS = 1;
+
+	/// @dev The number of virtual shares has been chosen low enough to prevent overflows, and high enough to ensure
+	/// high precision computations.
+	uint256 internal constant VIRTUAL_SHARES = 1e6;
+
+	enum Mode {
+		Collateral,
+		Repay,
+		Other // Neither collateral nor repay
+	}
+
+	/// @notice Handles Eth to Weth conversion if assets are provided.
+	function _performEthToWethConversion(
+		MarketParams memory _marketParams,
+		uint256 _assets,
+		address _onBehalf,
+		uint256 _getId,
+		Mode _mode
+	) internal returns (MarketParams memory, uint256 _amt) {
+		_amt = getUint(_getId, _assets);
+
+		bool _isEth = _mode == Mode.Collateral
+			? _marketParams.collateralToken == ethAddr
+			: _marketParams.loanToken == ethAddr;
+
+		_marketParams = updateTokenAddresses(_marketParams);
+
+		// Check for max value
+		if (_assets == type(uint256).max) {
+			uint256 _maxAvailable = _isEth
+				? address(this).balance
+				: TokenInterface(_marketParams.loanToken).balanceOf(
+					address(this)
+				);
+			if (_mode == Mode.Repay) {
+				uint256 _amtDebt = getPaybackBalance(_marketParams, _onBehalf);
+				_amt = min(_maxAvailable, _amtDebt);
+			} else {
+				_amt = _maxAvailable;
+			}
+		}
+
+		// Perform conversion if necessary
+		if (_isEth) {
+			convertEthToWeth(
```

Comment:
The value of `_isEth` is checked twice: once here and once inside `convertEthToWeth`

Reply:
Good catch! Removed the external check. [Here](https://github.com/Instadapp/dsa-connectors-2.0/pull/1/files#diff-57afa2d25dc241f0573736f488fd0f69847c6dfeeffa95efbdd7bc0fe9fec13eR83-R90)